File: java/poor-pigs.java
//
//
package java;

class PoorPigs {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int times = minutesToTest / minutesToDie;
        int base = times + 1;
        return (int) Math.ceil(Math.log(buckets) / Math.log(base));
    }
}

File: go/poor-pigs.go
//
//
package leetcode

import "math"

func poorPigs(buckets int, minutesToDie int, minutesToTest int) int {
	times := minutesToTest / minutesToDie
	base := times + 1
	log := math.Log(float64(buckets)) / math.Log(float64(base))
	return int(math.Ceil(log))
}
